ibis/expr/datatypes/core.py

```diff
@@ -602,6 +602,9 @@ def from_unit(cls, unit, timezone=None, nullable=True):
         elif unit == TimestampUnit.NANOSECOND:
             scale = 9
         else:
+            # TODO: remove raise path as it's never triggered
+            # Timestamp op has a restriction that only the literal
+            # ints from 0 through 10 can be passed as scale
```

Comment:
I think it should be 0 to 9, hopefully we're not allowing subnanoseconds 😬

Reply:
Whoops, nice catch.